src/partition/repart.c

```diff
@@ -2736,13 +2778,30 @@ static int do_copy_files(Partition *p, const char *fs) {
         return 0;
 }
 
-static int partition_copy_files(Partition *p, const char *node) {
+static int do_make_directories(Partition *p, const char *fs) {
+        char **d;
+        int r;
+
+        assert(p);
+        assert(fs);
+
+        STRV_FOREACH(d, p->make_directories) {
+
+                r = mkdir_p_root(fs, *d, UID_INVALID, GID_INVALID, 0755);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to create directory '%s' in file system: %m", *d);
```

Comment:
maybe include `fs` in the log message?

Reply:
It's an internal dir we mount stuff to temporarily inside a namespace, it shouldn't be shown to the user since it's not visible to them, and not accessible or persistent